test/functional/feature_presegwit_node_upgrade.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test a pre-segwit node upgrading to segwit consensus"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    softfork_active,
+)
+
+class SegwitUpgradeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.test_upgrade_after_activation()
+
+    def test_upgrade_after_activation(self):
+        """A pre-segwit node with insufficiently validated blocks needs to redownload blocks"""
```

Comment:
No need to have a run_test() method that only calls one method. Just put the logic in `run_test()`.

```suggestion
    def run_test(self):
        """A pre-segwit node with insufficiently validated blocks needs to redownload blocks"""
```

Reply:
I am not sure I understand. It seems we call `main()` which then calls `test_framework.py::run_test()`. I am not sure how I could remove this `run_test()` and put the logic in `test_framework.py::run_test()`? 

Also, it seems `BitcoinTestFramework` [requires](https://github.com/bitcoin/bitcoin/blob/480bf01c295527bd212964efe4df3bb886db5654/test/functional/test_framework/test_framework.py#L69) overriding `run_test()`.

Reply:
Have you tried applying this diff?

Reply:
You're right. My apologies for misreading the diff as just removing the `run_test()` function (which I did try manually). Updated. Thanks for your patience.